Lib/concurrent/futures/process.py

```diff
@@ -380,7 +389,9 @@ def wait_result_broken_or_wakeup(self):
 
         elif wakeup_reader in ready:
             is_broken = False
-        self.thread_wakeup.clear()
+
+        with self.shutdown_lock:
```

Comment:
I don't think this lock is necessary as the goal is to protect against close which is only called in the same thread.